### ISSUE TITLE
Autocomplete search breaking changes in 1.1.1 (Fix Debounce)

### ISSUE THREAD

**Issue by daywee** [Megabit/Blazorise#4185]
**Describe the bug**
After update from 1.0.4 to 1.1.1 I observed regression for some features.
1. Using a `SelectedValue` doesn't change current value. Current value is not updated after initialization, nor after clicking a `Change to last` button.
2. After clearing text with keyboard, the opened dropdown does not show all available items, it shows just the item that was here before search deletion.

**To Reproduce**
```razor
<Autocomplete TItem="AutocompleteItem"
              TValue="string"
              Data="_items"
              TextField="@(e => e.Name)"
              ValueField="@(e => e.Id)"
              SelectedValue="_selectedItemId"
              SelectedValueChanged="HandleItemIdChanged"
              SelectedText="@_items.SingleOrDefault(e => e.Id == _selectedItemId)?.Name"
              Placeholder="Search..."
              Filter="AutocompleteFilter.Contains"
              MinLength="0"
              MaxMenuHeight="450px"
              FreeTyping="false">
    <NotFoundContent>Item not found :(</NotFoundContent>
    <ItemContent>
        @context.Item.Name
    </ItemContent>
</Autocomplete>

<Button Clicked="@(() => _selectedItemId = _items.Last().Id )" Color="Color.Primary">Change to last</Button>
```
```csharp
public partial class AutocompletePage
{
    private List<AutocompleteItem> _items;
    private string _selectedItemId;

    protected override Task OnInitializedAsync()
    {
        _items = new List<AutocompleteItem>
        {
            new AutocompleteItem { Id = "first", Name = "FIRST" },
            new AutocompleteItem { Id = "second", Name = "SECOND" },
            new AutocompleteItem { Id = "third", Name = "THIRD" },
        };
        _selectedItemId = _items.First().Id;

        return base.OnInitializedAsync();
    }

    private Task HandleItemIdChanged(string id)
    {
        _selectedItemId = id;

        // do some other stuff

        return Task.CompletedTask;
    }
}

public class AutocompleteItem
{
    public string Id { get; set; }
    public string Name { get; set; }
}
```

**Expected behavior**
1. Autocomplete should have item set after initialization or after clicking a button.
![image](https://user-images.githubusercontent.com/13921707/194599116-00e11db0-6a19-45b4-9b15-1d1d92ce6a83.png)
2. After deleting text, the dropdown should show all items, not just last active.
![image](https://user-images.githubusercontent.com/13921707/194599306-9ecbb2c7-e6d4-482d-a769-a5273fca5f1a.png)

**Comment by daywee** [comment]
And I found third difference.

The background color of active item is gray now. By active item I mean selected by keyboard for example, hovered items are gray in both cases.
![image](https://user-images.githubusercontent.com/13921707/194707468-f6d3b5f5-8689-469d-9373-4bf0e31e56a5.png)


In 1.0.4, the primary color was used for that.
![image](https://user-images.githubusercontent.com/13921707/194707612-e5a8a0c0-9fa0-4532-9c53-5bc027b86bc7.png)

**Comment by David-Moreira** [comment]
> And I found third difference.
> 
> The background color of active item is gray now. By active item I mean selected by keyboard for example, hovered items are gray in both cases. ![image](https://user-images.githubusercontent.com/13921707/194707468-f6d3b5f5-8689-469d-9373-4bf0e31e56a5.png)
> 
> In 1.0.4, the primary color was used for that. ![image](https://user-images.githubusercontent.com/13921707/194707612-e5a8a0c0-9fa0-4532-9c53-5bc027b86bc7.png)

Hello,
This change was intended in order to differentiante actually selected items and "pre-selected" items and also as a way to make the components have more similar looks overall since generally the hover / pre selection is greyish or of a secondary theme color typically. 
**Dropdown List for example (BS4) (which is actually what autocomplete uses internally)**
![image](https://user-images.githubusercontent.com/22283161/194717252-7364d98c-6ce7-41b0-8053-85ff407609be.png)

**Actual need for this change (BS4):**
![image](https://user-images.githubusercontent.com/22283161/194717237-5ece258a-a156-4896-b8a5-f5c31fb04419.png)

Now I understand you might want the old look back in spite of our direction. In that case you should be able to use the following css:
```
.b-is-autocomplete .dropdown-menu .dropdown-item.focus {
    color: white !important;
    background-color: #0288D1 !important;
}
```

As for the other cases we'll take a look, it seems that it should work as you've described.

**Comment by David-Moreira** [comment]
> After clearing text with keyboard, the opened dropdown does not show all available items, it shows just the item that was here before search deletion.

Can you give us more detailed steps on this one? Including WASM or SERVER and provider?

I was not able to reproduce it, I get the following behaviour which seems fine:
![chrome-capture-2022-9-8](https://user-images.githubusercontent.com/22283161/194718287-afac2777-43e0-4dc5-a5bf-dbbe04b53f7f.gif)

**Comment by David-Moreira** [comment]
> Using a SelectedValue doesn't change current value. Current value is not updated after initialization, nor after clicking a Change to last button.

Any reason why you have this redudant conflicting line? 
`SelectedText="@_items.SingleOrDefault(e => e.Id == _selectedItemId)?.Name"`

By removing it, it works ok. Starts Selected and changes on button click as expected.
![chrome-capture-2022-9-8 (1)](https://user-images.githubusercontent.com/22283161/194718688-08302345-95a0-4e4d-b47c-13ab97ffb74d.gif)

**Comment by daywee** [comment]
Hi,

I'm using SERVER with Bootstrap.

Nice catch about the redundant line `SelectedText="@_items.SingleOrDefault(e => e.Id == _selectedItemId)?.Name"`, removing the line fixes the first problem.

About the third styling problem... Thank you for explanation and you are right, this change makes the look more similar to other components, which is great.

I investigated more the second problem with not showing all items when the input is cleared by backspace, and realized that I have `Debounce` enabled globally. Disabling it globally or on component directly fixes the issue. I also tried this directly in `Blazorise.Demo.Bootstrap.Server`, `Blazorise.Demo.Bootstrap` and `Blazorise.Demo.Bootstrap5` projects on commit 5ed9246e4b942c91aebf282afb1a9c4d83f81a9c (current master) and in all three cases the issue is here. So by enabling the debounce, you should be able to reproduce the issue now. Also, hitting an escape to clear it works perfectly in both cases.

**Comment by David-Moreira** [comment]
`Debouce` is definetly something we have to optimize with `Autocomplete`, there might be some unintended race conditions as `Autocomplete` as grown to be quite a complex component, at the moment I will have to advise you to disable it for the time being. It should provide a better experience when using the component.

We'll let you know when we optimize this parameter with  `Autocomplete`

**Comment by David-Moreira** [comment]
Closes through #4249
